tensorboard/components/vz_line_chart2/vz-line-chart2.ts

```diff
@@ -266,6 +266,53 @@ Polymer({
     '_outliersChanged(ignoreYOutliers, _chart)'
   ],
 
+  ready: function() {
+    this.scopeSubtree(this.$.tooltip, true);
+    this.scopeSubtree(this.$.chartdiv, true);
+  },
+
+  attached() {
+    const option = {capture: true, passive: true};
+    this._listen(this, 'mousedown', this._onMouseDown.bind(this), option);
+    this._listen(this, 'mouseup', this._onMouseUp.bind(this), option);
+    this._listen(window, 'keydown', this._onKeyDown.bind(this), option);
+    this._listen(window, 'keyup', this._onKeyUp.bind(this), option);
+  },
+
+  detached() {
+    this.cancelAsync(this._makeChartAsyncCallbackId);
+    if (this._chart) this._chart.destroy();
+    if (this._listeners) {
+      this._listeners.forEach(({node, eventName, func, option}) => {
+        node.removeEventListener(eventName, func, option);
+      });
+      this._listeners.clear();
+    }
+  },
+
+  _listen(node: Node, eventName: string, func: (event) => void, option = {}) {
+    if (!this._listeners) this._listeners = new Set();
+    this._listeners.add({node, eventName, func, option});
+    node.addEventListener(eventName, func, option);
+  },
+
+
+  _onKeyDown(event) {
+    this.toggleClass('altdown', event.altKey);
+  },
+
+  _onKeyUp(event) {
+    this.toggleClass('altdown', false);
```

Comment:
Shouldn't the value to set here be event.altKey again?

Otherwise, in my reading of the code, any `keyup` event will turn off the aldown class, even if the key released wasn't the alt key.  E.g. if you 1) press `alt`, 2) press `shift, 3) release `shift`, then 4) release `alt`, the altdown class should only be removed at step 4, but I think this logic will remove it at step 3.